### PR TITLE
Memory ops for RVSDG => Bril

### LINKED ISSUE #343: RVSDG with memory -> Bril translation

### PR DESCRIPTION
**Pull request by ezrosent**
Wire in memory ops in to_cfg. Opening a draft as there are a few questions in to_cfg I'd like to resolve.

Fixes #343

### CODE REVIEW

File: src/rvsdg/to_cfg.rs [hunk 1]
```diff
@@ -763,33 +797,35 @@ impl<'a> RvsdgToCfg<'a> {
...
-                assert!(print_operands.len() == 2);
-                let argument = self.operand_to_bril(print_operands[0], current_args, ctx);
-                // also need to evaluate other prints before this one
-                let second_evaluated = self.operand_to_bril(print_operands[1], current_args, ctx);
-
-                let instructions = vec![Instruction::Effect {
-                    op: EffectOps::Print,
-                    args: vec![argument.get_single_res().unwrap_name()],
+            BasicExpr::Effect(op, args) => {
+                // TODO: should we combine these?
```
Comment (oflatt) [inline src/rvsdg/to_cfg.rs:801]: remove todo, comment that we combine all but the state edge

**Review by oflatt — CHANGES_REQUESTED** [review]
Nice work!
Is this tested by the current infra? I think you need to say that the RVSDG roundtrip run mode supports memory.

**Comment by ezrosent** [comment]
Done! This uncovered a preexisting issue with converting empty functions:

```
@funcA(p: ptr<int>) {
  ret;
}

@main {
  five: int = const 5;
  x: ptr<int> = alloc five;
  call @funcA x;
  free x;
}
```

Gets translated back correctly, but `@funcA` had an empty body

```
@funcA(p: ptr<int>) {
}
```

Which caused the bril interpreter to panic. I added some code to fix that.

**Review by oflatt — APPROVED** [review]
Great!

**Pull request merged**